app/_hub/kong-inc/upstream-oauth/overview/_index.md

```diff
@@ -0,0 +1,76 @@
+---
+nav_title: Overview
+title: Overview
+---
+
+The Upstream OAuth plugin allows {{site.base_gateway}} to support OAuth flows between Kong and the upstream API.
+
+The plugin supports storing tokens issued by the IdP in different backend formats.
+
+## How it works
+
+The upstream OAuth2 credential flow works similarly to the [client credentials grant](/hub/kong-inc/openid-connect/how-to/authentication/client-credentials/) used by the OpenID Connect plugin. If a cached access token isn't found, Kong issues a request to the IdP token endpoint to obtain a new token, which is cached, and then passed to the upstream API via a configurable header (`Authorization` by default).
+
+<!--vale off-->
+
+{% mermaid %}
+sequenceDiagram
+    autonumber
+    participant client as Client <br>(e.g. mobile app)
+    participant kong as API Gateway <br>(Kong)
+    participant idp as IDP <br>(e.g. Keycloak)
+    participant api as 3rd Party API
+    activate client
+    activate kong
+    client->>kong: request to Kong
+    deactivate client
+    activate idp
+    kong->>idp: request access token <br>from IdP using <br>client ID and client secret (if IdP auth is set)
+    deactivate kong
+    idp->>idp: authenticate client
+    activate kong
+    idp->>kong: return access token
+    deactivate idp
+    activate api
+    kong->>api: request with access token in <br>Authorization header
+    deactivate kong
+    activate kong
+    api->>kong: response
+    deactivate api
+    activate client
+    kong->>client: response
+    deactivate client
+    deactivate kong
+{% endmermaid %}
+
+<!--vale on-->
+
+
+## Authentication methods
+
+This plugin supports the following authentication methods:
+
+* [`client_secret_basic`](/hub/kong-inc/upstream-oauth/configuration/#config-client_secret_basic): Send `client_id` and `client_secret` in an `Authorization: Basic` header
+* [`client_secret_post`](/hub/kong-inc/upstream-oauth/configuration/#config-client_secret_basic): Send `client_id` and `client_secret` as part of the body
+* [`client_secret_jwt`](/hub/kong-inc/upstream-oauth/configuration/#config-client_secret_basic): Send a JWT signed with the `client_secret` using the client assertion as part of the body
+
+## Caching
+
+The Upstream OAuth plugin caches tokens returned by the IdP.  
+Cache entries expire based on the `expires_in` indicated by the IdP in the response to the token endpoint.
```

Comment:
```suggestion
Cached entries expire based on the `expires_in` indicated by the IdP in the response to the token endpoint.
```
cached because the entries are already cached?